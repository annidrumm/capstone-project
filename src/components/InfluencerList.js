import React from 'react'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

export default function InfluencerList({ influencerData }) {
  return (
    <ContainerStyled>
      <section className="posts">
        {influencerData.map((post) => {
          return (
            <section key={post.id}>
              <Container>
                <Username>{post.username}</Username>
                <Categories>{post.categories}</Categories>
              </Container>
              <DetailList>
                <DetailItem key={post.username}>
                  Follower: {post.follower}
                </DetailItem>
                <DetailItem key={post.gender}>Gener: {post.gender}</DetailItem>
                <DetailItem key={post.age}>Age: {post.age}</DetailItem>
                <DetailItem key={post.location}>
                  Location: {post.location}
                </DetailItem>
                <DetailItem key={post.languages}>
                  Languages: {post.languages}{' '}
                </DetailItem>
              </DetailList>
            </section>
          )
        })}
      </section>
    </ContainerStyled>
  )
}

const ContainerStyled = styled.section`
  width: 400px;
`

const Container = styled.section`
  background: #ffffff1a;
  height: 200px;
  width: 400px;
`
const Username = styled.h2`
  font-family: 'Poppins';
  font-weight: 600;
  line-height: 100px;
  text-align: center;
`
const Categories = styled.h3`
  font-family: 'Poppins';
  font-weight: 300;
  text-align: center;
`

const DetailList = styled.ul`
  text-align: left;
  list-style-type: none;
  font-family: 'Poppins';
  font-weight: 300;
`
const DetailItem = styled.li`
  margin: 10px 0;
`

InfluencerList.propTypes = {
  influencerData: PropTypes.array,
}
