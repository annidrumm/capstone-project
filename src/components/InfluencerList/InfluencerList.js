import React from 'react'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

export default function InfluencerList({ influencerData }) {
  return (
    <section className="profiles">
      {influencerData.map((profile) => {
        return (
          <section key={profile.id}>
            <ContainerStyled>
              <UsernameStyled>{profile.username}</UsernameStyled>
              <CategoriesStyled>{profile.categories}</CategoriesStyled>
            </ContainerStyled>
            <DetailListStyled>
              <DetailItemStyled key={profile.username}>
                Follower: {profile.follower}
              </DetailItemStyled>
              <DetailItemStyled key={profile.gender}>
                Gender: {profile.gender}
              </DetailItemStyled>
              <DetailItemStyled key={profile.age}>
                Age: {profile.age}
              </DetailItemStyled>
              <DetailItemStyled key={profile.location}>
                Location: {profile.location}
              </DetailItemStyled>
              <DetailItemStyled key={profile.languages}>
                Languages: {profile.languages}{' '}
              </DetailItemStyled>
            </DetailListStyled>
          </section>
        )
      })}
    </section>
  )
}

const ContainerStyled = styled.section`
  background: #ffffff1a;
  height: 200px;
  width: 100%;
`
const UsernameStyled = styled.h2`
  font-family: 'Poppins';
  font-weight: 600;
  line-height: 100px;
  text-align: center;
`
const CategoriesStyled = styled.h3`
  font-family: 'Poppins';
  font-weight: 300;
  text-align: center;
`

const DetailListStyled = styled.ul`
  text-align: left;
  list-style-type: none;
  font-family: 'Poppins';
  font-weight: 300;
  width: 100%;
`
const DetailItemStyled = styled.li`
  margin: 10px 0;
  position: relative;
  left: 40px;
`

InfluencerList.propTypes = {
  influencerData: PropTypes.array,
}
