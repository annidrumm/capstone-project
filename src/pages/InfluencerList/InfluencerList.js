import React from 'react'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import Headline from '../../components/Headline/Headline'
import backIcon from '../../icons/back.svg'
import { useHistory } from 'react-router-dom'

export default function InfluencerList({ influencerData }) {
  const history = useHistory()
  const routeChange = () => {
    let path = `/`
    history.push(path)
  }

  return (
    <section className="profiles">
      <GoBackStyled src={backIcon} alt="arrow left" onClick={routeChange} />
      <Headline headline="Search" />
      {influencerData.map((profile) => {
        return (
          <section key={profile.id}>
            <ContainerStyled>
              <UsernameStyled>{profile.username}</UsernameStyled>
              <CategoriesStyled>{profile.categories}</CategoriesStyled>
            </ContainerStyled>
            <DetailListStyled>
              <DetailItemStyled key={profile.username}>
                Follower: {profile.follower}
              </DetailItemStyled>
              <DetailItemStyled key={profile.gender}>
                Gender: {profile.gender}
              </DetailItemStyled>
              <DetailItemStyled key={profile.age}>
                Age: {profile.age}
              </DetailItemStyled>
              <DetailItemStyled key={profile.location}>
                Location: {profile.location}
              </DetailItemStyled>
              <DetailItemStyled key={profile.languages}>
                Languages: {profile.languages}{' '}
              </DetailItemStyled>
            </DetailListStyled>
          </section>
        )
      })}
    </section>
  )
}

const ContainerStyled = styled.section`
  background: var(--transparent-blue);
  height: 200px;
  width: 100%;
`
const GoBackStyled = styled.img`
  position: absolute;
  left: 30px;
  margin-top: 10px;
  cursor: pointer;
`

const UsernameStyled = styled.h2`
  font-family: 'Poppins';
  font-weight: var(--font-weight-semibold);
  line-height: 100px;
  text-align: center;
`
const CategoriesStyled = styled.h3`
  font-family: 'Poppins';
  font-weight: var(--font-weight-light);
  text-align: center;
`

const DetailListStyled = styled.ul`
  text-align: left;
  list-style-type: none;
  font-family: 'Poppins';
  font-weight: var(--font-weight-light);
  width: 100%;
  margin: 50px;
`
const DetailItemStyled = styled.li`
  letter-spacing: 1px;
  margin: 10px 0;
  position: relative;
`

InfluencerList.propTypes = {
  influencerData: PropTypes.array,
}
