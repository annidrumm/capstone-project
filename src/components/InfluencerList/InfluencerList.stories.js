import React from 'react'
import InfluencerList from './InfluencerList'
import influencerData from '../../mocks/influencer.json'

export default {
  title: 'InfluencerList',
  component: InfluencerList,
}

export const Influencer = () => {
  return <InfluencerList influencerData={influencerData} />
}
