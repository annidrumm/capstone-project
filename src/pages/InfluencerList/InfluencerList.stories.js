import React from 'react'
import InfluencerList from './InfluencerList'
import influencerData from '../../mocks/influencer.json'

export default {
  title: 'Pages/InfluencerList',
  component: InfluencerList,
}

export const fullList = () => {
  return <InfluencerList influencerData={influencerData} />
}
