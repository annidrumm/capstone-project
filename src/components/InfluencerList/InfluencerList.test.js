import React from 'react'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import InfluencerList from './InfluencerList.js'
import 'jest-styled-components'

describe('InfluencerList', () => {
  const influencerData = [
    {
      id: 1,
      username: '@fitness4life',
      location: 'Hamburg',
      age: '25 years',
    },
    {
      id: 2,
      username: '@VeganPlace',
      location: 'Munich',
      age: '28 years',
    },
  ]

  it('display the username from a influencer', () => {
    const { getByText } = render(
      <InfluencerList influencerData={influencerData} />
    )
    influencerData.forEach(({ username, location, age }) => {
      expect(getByText(username)).toBeInTheDocument()
      expect(getByText(new RegExp(age, 'i'))).toBeInTheDocument()
      expect(getByText(new RegExp(location, 'i'))).toBeInTheDocument()
    })
  })

  it('renders username correctly', () => {
    const tree = renderer.create(
      <InfluencerList influencerData={influencerData} />
    )
    expect(tree).toMatchSnapshot()
  })
})
