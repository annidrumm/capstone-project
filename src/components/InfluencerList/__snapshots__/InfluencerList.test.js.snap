// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfluencerList renders username correctly 1`] = `
.c0 {
  background: #ffffff1a;
  height: 200px;
  width: 100%;
}

.c1 {
  font-family: 'Poppins';
  font-weight: 600;
  line-height: 100px;
  text-align: center;
}

.c2 {
  font-family: 'Poppins';
  font-weight: 300;
  text-align: center;
}

.c3 {
  text-align: left;
  list-style-type: none;
  font-family: 'Poppins';
  font-weight: 300;
  width: 100%;
}

.c4 {
  margin: 10px 0;
  position: relative;
  left: 40px;
}

<section
  className="profiles"
>
  <section>
    <section
      className="c0"
    >
      <h2
        className="c1"
      >
        @fitness4life
      </h2>
      <h3
        className="c2"
      >
        Fitness
      </h3>
    </section>
    <ul
      className="c3"
    >
      <li
        className="c4"
      >
        Follower: 
        10k
      </li>
      <li
        className="c4"
      >
        Gender: 
        female
      </li>
      <li
        className="c4"
      >
        Age: 
        25 years
      </li>
      <li
        className="c4"
      >
        Location: 
        Hamburg
      </li>
      <li
        className="c4"
      >
        Languages: 
        German, 
        English
         
      </li>
    </ul>
  </section>
  <section>
    <section
      className="c0"
    >
      <h2
        className="c1"
      >
        @VeganPlace
      </h2>
      <h3
        className="c2"
      >
        Food
        Health
      </h3>
    </section>
    <ul
      className="c3"
    >
      <li
        className="c4"
      >
        Follower: 
        1.000
      </li>
      <li
        className="c4"
      >
        Gender: 
        female
      </li>
      <li
        className="c4"
      >
        Age: 
        28 years
      </li>
      <li
        className="c4"
      >
        Location: 
        Munich
      </li>
      <li
        className="c4"
      >
        Languages: 
        German, 
        English
         
      </li>
    </ul>
  </section>
</section>
`;
